ContactMechanics/FFTElasticHalfSpace.py

```diff
@@ -744,8 +703,9 @@ def _compute_fourier_coeffs(self):
                                                             (x_s - a) * (x_s - a))) /  # noqa: E501
                                        ((x_s + a) + np.sqrt((y_s - b) * (y_s - b) +  # noqa: E501
                                                             (x_s + a) * (x_s + a)))))  # noqa: E501
-            self.fftengine.fft(facts, self.weights)
-            return self.weights, facts
+            self.fftengine.fft(self.real_buffer, self.fourier_buffer)
+            self.weights = self.fourier_buffer.array().copy()
+            return self.weights, self.real_buffer.array()
```

Comment:
Yes, that is correct. b is never used. I wanted to avoid the copy here. That said, self.real_buffer may be destroyed.

Reply:
This was exactly the failure...

Reply:
you mean that real_buffer is destroyed during the fft because it is doing some in-place stuff ? 

If you want to get rid of the copy, I think we can stop testing facts in the tests. It is redundant with testing weights anyway.

Reply:
Yes, I'm in the process of removing the facts. I will also rename `weights` to `greens_function`.